src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,329 @@
+
+            dswx_sar_util.write_raster_block(
+                fuzzy_map_temp,
+                region_grow_block,
+                block_param,
+                geotransform=meta_dict['geotransform'],
+                projection=meta_dict['projection'],
+                DataType='float32')
+
+            # In final loop, write the result to output_tif_path
+            if loopind == num_loop - 1:
+                dswx_sar_util.write_raster_block(
+                    output_tif_path,
+                    region_grow_block,
+                    block_param,
+                    geotransform=meta_dict['geotransform'],
+                    projection=meta_dict['projection'])
+
+
+def run(cfg):
+    '''
+    Run region growing with parameters in cfg dictionary
+    '''
+    logger.info('Starting DSWx-S1 Region-Growing')
+
+    t_all = time.time()
+
+    processing_cfg = cfg.groups.processing
+    outputdir = cfg.groups.product_path_group.scratch_path
+    pol_str = '_'.join(processing_cfg.polarizations)
+
+    # Region growing cfg
+    region_growing_cfg = processing_cfg.region_growing
+    region_growing_seed = region_growing_cfg.initial_seed
+    region_growing_relaxed_threshold = region_growing_cfg.relaxed_threshold
+    region_growing_line_per_block = region_growing_cfg.line_per_block
+
+    print(f'Region Growing Seed: {region_growing_seed}')
+    print(f'Region Growing relaxed threshold: {region_growing_relaxed_threshold}')
+
+    fuzzy_tif_path = os.path.join(outputdir,
+                                  f'fuzzy_image_{pol_str}.tif')
+    water_meta = dswx_sar_util.get_meta_from_tif(fuzzy_tif_path)
+    water_tif_path = os.path.join(outputdir,
+                                  f"region_growing_output_binary_{pol_str}.tif")
+    temp_rg_tif_path = os.path.join(outputdir,
+                                    f'temp_region_growing_{pol_str}.tif')
+
+    # First, run region-growing algorithm for blocks
+    # to avoid to repeatly run with large image.
+    run_parallel_region_growing(
+        fuzzy_tif_path,
+        temp_rg_tif_path,
+        lines_per_block=region_growing_line_per_block,
+        initial_seed=region_growing_seed,
+        relaxed_threshold=region_growing_relaxed_threshold,
+        maxiter=0)
+
+    fuzzy_map = dswx_sar_util.read_geotiff(fuzzy_tif_path)
+    temp_rg = dswx_sar_util.read_geotiff(temp_rg_tif_path)
+
+    # replace the fuzzy values with 1 for the pixels
+    # where the region-growing already applied
+    fuzzy_map[temp_rg == 1] = 1
+
+    # Run region-growing again for entire image
+    region_grow_map = region_growing(
+        fuzzy_map,
+        initial_seed=region_growing_seed,
+        relaxed_threshold=region_growing_relaxed_threshold,
+        maxiter=0)
+
+    dswx_sar_util.save_dswx_product(
+        region_grow_map,
+        water_tif_path,
+        geotransform=water_meta['geotransform'],
+        projection=water_meta['projection'],
+        description='Water classification (WTR)',
```

Comment:
Not sure what to do here with `Water`. Is running region_growing.py meant specifically water and supporting functions meant to be generalized any feature?

Reply:
The product is related to the water map, but it doesn't need to include the description here. I'll delete the description. Thank you for pointing out.